mmselfsup/models/backbones/resnext.py

```diff
@@ -1,165 +1,56 @@
-
-        self.norm1_name, norm1 = build_norm_layer(
-            self.norm_cfg, width, postfix=1)
-        self.norm2_name, norm2 = build_norm_layer(
-            self.norm_cfg, width, postfix=2)
-        self.norm3_name, norm3 = build_norm_layer(
-            self.norm_cfg, self.planes * self.expansion, postfix=3)
-
-        self.conv1 = build_conv_layer(
-            self.conv_cfg,
-            self.inplanes,
-            width,
-            kernel_size=1,
-            stride=self.conv1_stride,
-            bias=False)
-        self.add_module(self.norm1_name, norm1)
-
-        self.conv2 = build_conv_layer(
-            self.conv_cfg,
-            width,
-            width,
-            kernel_size=3,
-            stride=self.conv2_stride,
-            padding=self.dilation,
-            dilation=self.dilation,
-            groups=groups,
-            bias=False)
-        self.add_module(self.norm2_name, norm2)
-
-        self.conv3 = build_conv_layer(
-            self.conv_cfg,
-            width,
-            self.planes * self.expansion,
-            kernel_size=1,
-            bias=False)
-        self.add_module(self.norm3_name, norm3)
-
-
-def make_res_layer(block,
-                   inplanes,
-                   planes,
-                   blocks,
-                   stride=1,
-                   dilation=1,
-                   groups=1,
-                   width_per_group=4,
-                   style='pytorch',
-                   with_cp=False,
-                   conv_cfg=None,
-                   norm_cfg=dict(type='BN')):
-    downsample = None
-    if stride != 1 or inplanes != planes * block.expansion:
-        downsample = nn.Sequential(
-            build_conv_layer(
-                conv_cfg,
-                inplanes,
-                planes * block.expansion,
-                kernel_size=1,
-                stride=stride,
-                bias=False),
-            build_norm_layer(norm_cfg, planes * block.expansion)[1],
-        )
-
-    layers = []
-    layers.append(
-        block(
-            inplanes=inplanes,
-            planes=planes,
-            stride=stride,
-            dilation=dilation,
-            downsample=downsample,
-            groups=groups,
-            width_per_group=width_per_group,
-            style=style,
-            with_cp=with_cp,
-            conv_cfg=conv_cfg,
-            norm_cfg=norm_cfg))
-    inplanes = planes * block.expansion
-    for i in range(1, blocks):
-        layers.append(
-            block(
-                inplanes=inplanes,
-                planes=planes,
-                stride=1,
-                dilation=dilation,
-                groups=groups,
-                width_per_group=width_per_group,
-                style=style,
-                with_cp=with_cp,
-                conv_cfg=conv_cfg,
-                norm_cfg=norm_cfg))
-
-    return nn.Sequential(*layers)
-
-
 @BACKBONES.register_module()
```

Comment:
Why do we still need ResNeXt? Can we also import `mmcls.ResNeXt`?

Reply:
> Why do we still need ResNeXt? Can we also import `mmcls.ResNeXt`?

As we are using forward function from our own ResNet, which is different from forward function in mmcls.

Reply:
> Why do we still need ResNeXt? Can we also import `mmcls.ResNeXt`?

As we are using forward function from our own ResNet, which is different from forward function in mmcls.

Reply:
> Why do we still need ResNeXt? Can we also import `mmcls.ResNeXt`?

MMCls does not output the feature map from the first `conv` layer, which we will use for downstream evaluation. That's why we rewrite the `forward` function.

Reply:
1. We should add comments in the code to explain the difference and the reason. 
2. We can give feedback to mmcls, e.g., create a feature request. Invite @mzr1996 to take a look.